### PR TITLE
Use setCompressionQuality when setting image quality

### PR DESCRIPTION
**Pull request by adamsilverstein**
Trac ticket: https://core.trac.wordpress.org/ticket/61614

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: tests/phpunit/tests/media.php [hunk 1]
```diff
@@ -5283,11 +5283,30 @@ public function test_quality_with_image_conversion_file_sizes() {
...
+
+		// Repeat the size tests with AVIF images.
+		add_filter( 'image_editor_output_format', array( $this, 'image_editor_output_avif' ) );
+		$avif_sizes = wp_generate_attachment_metadata( $attachment_id, $file );
+		remove_filter( 'image_editor_output_format', array( $this, 'image_editor_output_avif' ) );
+
+		// Sub-sizes: for each size, the AVIF should be smaller than the JPEG.
+		$sizes_to_compare = array_intersect_key( $jpeg_sizes['sizes'], $avif_sizes['sizes'] );
+		foreach ( $sizes_to_compare as $size => $size_data ) {
+			$this->assertLessThan( $avif_sizes['sizes'][ $size ]['filesize'], $jpeg_sizes['sizes'][ $size ]['filesize'] );
```
Comment (mukeshpanchal27) [inline tests/phpunit/tests/media.php]: Something goes wrong with the assert. The reverse check is passed but why 🤔

**Review by mukeshpanchal27 — APPROVED** [review]
The code changes looks good to me. Left comment for unit tests

File: tests/phpunit/tests/media.php (hunk 1 above)
Comment (adamsilverstein) [reply tests/phpunit/tests/media.php]: thanks for reviewing. I'll take a look and make sure high quality is used for the first pass.

Comment (adamsilverstein) [reply tests/phpunit/tests/media.php]: 🤞🏼 fixed the unit tests in bf60efc7afab9e631db8324892d9599acc263905

**Comment by adamsilverstein** [comment]
Merged in https://core.trac.wordpress.org/changeset/59247

**Pull request closed without merging**